corrct/solvers.py

```diff
@@ -551,6 +551,164 @@ def __call__(  # noqa: C901
         return x, info
 
 
+class MLEM(Solver):
+    """
+    Initialize the MLEM solver class.
+
+    This class implements the Maximul Likelihood Expectation Maximization (MLEM) algorithm.
+
+    Parameters
+    ----------
+    verbose : bool, optional
+        Turn on verbose output. The default is False.
+    tolerance : Optional[float], optional
+        Tolerance on the data residual for computing when to stop iterations.
+        The default is None.
+    relaxation : float, optional
+        The relaxation length. The default is 1.95.
+    regularizer : Sequence[BaseRegularizer] | BaseRegularizer | None, optional
+        Regularizer to be used. The default is None.
+    data_term : Union[str, DataFidelityBase], optional
+        Data fidelity term for computing the data residual. The default is "l2".
+    data_term_test : Optional[DataFidelityBase], optional
+        The data fidelity to be used for the test set.
+        If None, it will use the same as for the rest of the data.
+        The default is None.
+    """
+
+    def __init__(
+        self,
+        verbose: bool = False,
+        relaxation: float = 1.95,
+        tolerance: Optional[float] = None,
+        regularizer: Union[Sequence[BaseRegularizer], BaseRegularizer, None] = None,
+        data_term: Union[str, DataFidelityBase] = "l2",
+        data_term_test: Union[str, DataFidelityBase, None] = None,
+    ):
+        super().__init__(
+            verbose=verbose, relaxation=relaxation, tolerance=tolerance, data_term=data_term, data_term_test=data_term_test
+        )
+        self.regularizer = self._initialize_regularizer(regularizer)
+
+    def info(self) -> str:
+        """
+        Return the MLEM info.
+
+        Returns
+        -------
+        str
+                 info string.
+        """
+        reg_info = "".join(["-" + r.info().upper() for r in self.regularizer])
+        return Solver.info(self) + "-" + self.data_term.info() + reg_info
```

Comment:
This might lead to incorrect info, as the noise model is fixed and regularization should not be supported (at least for this first implementation iteration)